pyfr/writers/paraview.py

```diff
@@ -22,45 +26,66 @@ class ParaviewWriter(BaseWriter):
     }
 
     def write_out(self):
-        """Controls the writing of serial .vtu Paraview files
-
-        Writes .vtu pieces for each element type, in each partition of
-        the PyFR files.  The Paraview data type used is "appended",
-        which concatenates all data into a single block of binary data
-        at the end of the file.  ASCII headers written at the top of
-        the file describe the structure of this data.
-        """
+        extn = os.path.splitext(self.outf)[1]
+        self._write_out(extn == '.pvtu')
+
+    def _write_out(self, parallel):
         # Set default divisor to solution order
         if self.args.divisor == 0:
             self.args.divisor = self.cfg.getint('solver', 'order')
 
-        write_s = lambda s: self.outf.write(s.encode('utf-8'))
-
-        # Write .vtu file header
-        write_s('<?xml version="1.0" ?>\n<VTKFile '
-                'byte_order="LittleEndian" type="UnstructuredGrid" '
-                'version="0.1">\n<UnstructuredGrid>\n')
-
-        # Initialise offset (in bytes) to end of appended data
-        off = 0
-
-        # Write data description header.  A vtk "piece" is used for each
-        # element in a partition.
-        for mk, sk in zip(self.mesh_inf, self.soln_inf):
-            off += _write_vtu_header(self.args, self.outf, self.mesh_inf[mk],
-                                     self.soln_inf[sk], off)
-
-        # Write end/start of header/data sections
-        write_s('</UnstructuredGrid>\n<AppendedData encoding="raw">\n_')
+        basename = os.path.splitext(self.outf)[0]
```

Comment:
Variable name is possibly misleading as it is not a basename (in the sense of `os.path.basename`).